agent/internal/agent.go

```diff
@@ -121,46 +119,24 @@ func (a *Agent) run(ctx context.Context) error {
 	}
 
 	a.log.Tracef("setting up %s runtime", a.opts.ContainerRuntime)
+	if a.opts.ContainerRuntime != options.DockerContainerRuntime {
+		a.log.Error("%w creation is not supported, please update agent container runtime config to use Docker instead.",
+			a.opts.ContainerRuntime)
+		return fmt.Errorf("%s creation not available", a.opts.ContainerRuntime)
+	}
+
 	var cruntime container.ContainerRuntime
```

Comment:
I think we don't need to declare cruntime ahead of time

```
	var cruntime container.ContainerRuntime
```

and can just do this on line 139 
```
	cruntime := docker.NewClient(dcl)
```